packages/api/endpoints/dead-letter-archive.js

```diff
@@ -11,27 +11,39 @@ async function postRecoverCumulusMessages(req, res) {
   const stackName = process.env.stackName;
   const systemBucket = process.env.system_bucket;
 
-  const { bucket, path } = (req.body === undefined ? {} : req.body);
+  const {
+    bucket,
+    path,
+    batchSize = 1000,
+    concurrency = 30,
+    maxDbPool = 50,
+  } = req.body ?? {}; // TODO ZOD!  KNEEL BEFORE ZOD
   const asyncOperation = await asyncOperations.startAsyncOperation({
     cluster: process.env.EcsCluster,
     callerLambdaName: getFunctionNameFromRequestContext(req),
     lambdaName: process.env.DeadLetterProcessingLambda,
-    asyncOperationTaskDefinition: process.env.AsyncOperationTaskDefinition,
+    asyncOperationTaskDefinition: process.env.DeadLetterRecoveryTaskDefinition,
     description: 'Dead-Letter Processor ECS Run',
     operationType: 'Dead-Letter Processing',
     payload: {
+      batchSize,
       bucket,
+      concurrency,
       path,
     },
     stackName,
     systemBucket,
-    knexConfig: process.env,
+    knexConfig: { ...process.env, maxDbPool },
```

Comment:
why isn't this configured as an environment variable? I note in your tests you're testing it as set in the env

Reply:
In this instance to be explicit as to where it's coming from - in non test scenarios I think we should strongly prefer not modifying process.env directly.    @cumulus/db allows configuration defaults to be set via environment variables, this line is making it clear that value is being overridden by user input.

Reply:
Got it -  In this case the maxDbPool is referring to the connection pool that's part of the Knex object in this code.   That pool allocates DB connections across concurrent requests, which in theory functionally reduces the number of open connections needed and/or handles connection retries, etc. 

This value doesn't have anything to do with the DB cluster itself, but rather the DLA recovery invocation's limit on total connections it can use.    The value here should definitely be set such that it doesn't overwhelm the DB cluster.

Reply:
Where is the document about maxDbPool?
I see knex example configuration https://knexjs.org/guide/#pool `pool: { min: 0, max: 7 }`